providers/desec/desecProvider.go

```diff
@@ -0,0 +1,223 @@
+package desec
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/StackExchange/dnscontrol/v3/models"
+	"github.com/StackExchange/dnscontrol/v3/pkg/diff"
+	"github.com/StackExchange/dnscontrol/v3/pkg/printer"
+	"github.com/StackExchange/dnscontrol/v3/providers"
+	"github.com/miekg/dns/dnsutil"
+)
+
+/*
+desec API DNS provider:
+Info required in `creds.json`:
+   - auth-token
+*/
+
+// NewDeSec creates the provider.
+func NewDeSec(m map[string]string, metadata json.RawMessage) (providers.DNSServiceProvider, error) {
+	c := &api{}
+	c.creds.token = m["auth-token"]
+	if c.creds.token == "" {
+		return nil, fmt.Errorf("missing deSEC auth-token")
+	}
+
+	// Get a domain to validate authentication
+	if err := c.fetchDomainList(); err != nil {
+		return nil, err
+	}
+
+	return c, nil
+}
+
+var features = providers.DocumentationNotes{
+	providers.DocDualHost:            providers.Unimplemented(),
+	providers.DocOfficiallySupported: providers.Cannot(),
+	providers.DocCreateDomains:       providers.Can(),
+	providers.CanUseAlias:            providers.Cannot(),
+	providers.CanUseSRV:              providers.Can(),
+	providers.CanUseSSHFP:            providers.Cannot(),
+	providers.CanUseCAA:              providers.Can(),
+	providers.CanUseTLSA:             providers.Can(),
+	providers.CanUsePTR:              providers.Unimplemented(),
+	providers.CanGetZones:            providers.Can(),
+	providers.CanAutoDNSSEC:          providers.Cannot(),
```

Comment:
deSEC automatically signs, but not sure what AutoDESEC exactly means here.

Reply:
As far as i understood this CanAutoDNSSEC this would require deSEC to be the registrar not only the authoritive nameserver.

Reply:
AutoDNSSEC means that the DNS provider does all the DNSSEC for you (including holding your private key).

For the first release it is best to disable things if you are unsure about them, then enable them in future releases as they are tested.

Reply:
deSEC does handle all DNSSEC aspects, including holding the private key. That's one of the reasons why the project exists. :-)

Reply:
Oh!  That's why its called deSEC!  I get it now!

Reply:
I'm still not sure if we talk about the same definition of AutoDNSSEC.
deSEC does sign the zone and provide the DS Records to their users, but each user has to transmit this DS Records to their registrar so that they are added on tld level.
And deSEC does sign those zones per default, you do not need to explicitly request it!

For example i got my neuland.tech domain registered at get.tech directly, i can add this domain to deSEC and they sign the zone. But i will have to add the DS Records using the control panel of get.techt manually.

Is this the definition / meaning of this AutoDNSSEC of dnscontrol?

Reply:
I don't use the feature myself (yet!) but I think that's how most providers implement it.

Reply:
If someone creates the domain / zone via dns control he gets the DS stuff printed, thats how i implemented it in dnscontrol. But its nothing they have to ask for, its the default behavior.